molgenis-semantic-mapper/src/main/java/org/molgenis/data/mapper/repository/impl/MappingProjectRepositoryImpl.java

```diff
@@ -43,7 +44,7 @@ public void add(MappingProject mappingProject)
 	{
 		if (mappingProject.getIdentifier() != null)
 		{
-			throw new MolgenisDataException("MappingProject already exists");
+			throw new DuplicateMappingProjectException(mappingProject.getIdentifier());
```

Comment:
What happens if we simply call add for existing? Perhaps the more generic one gets thrown already

Reply:
The behavior for entities stored in a PostgreSQL repository is to throw a UniqueConstraintViolationException with English message like  ```Duplicate value '...' for unique attribute '...' from entity '...'.```.

Reply:
Let's re-evaluate this exception when looking at the über pull request. We'll probably decide to 'share' more than one exception in this manner.